modules/internal/String.chpl

```diff
@@ -1069,29 +1069,30 @@ module String {
         var readIdx = 0;
         var writeIdx = 0;
         while readIdx < localThis.buffLen {
-          var cp: int(32);
-          var nbytes: c_int;
           var multibytes = (localThis.buff + readIdx): c_string;
-          var maxbytes = (localThis.buffLen - readIdx): ssize_t;
-          const decodeRet = qio_decode_char_buf_esc(cp, nbytes, multibytes,
-                                                    maxbytes);
+          var allowEsc = true;
+          
+          const (decodeRet, cp, nBytes) =  decodeHelp(localThis.buff, 
+                                                       localThis.buffLen,
+                                                       readIdx, allowEsc);
+          
           if (0xdc80<=cp && cp<=0xdcff) {
-            buff[writeIdx] = (cp-0xdc00):byteType;
+            buf[writeIdx] = (cp-0xdc00):byteType;
             writeIdx += 1;
           }
-          else if (decodeRet != 0) {
-            // the string contains invalid data
-            // at this point this can only happen due to a failure in our
-            // implementation of string encoding/decoding
-            // simply copy the data out
-            bufferMemcpyLocal(dst=(buff+writeIdx), src=multibytes, len=nbytes);
-            writeIdx += nbytes;
+          else if (decodeRet!=0){
+          // if the string contains invalid data
```

Comment:
I understand why to added this but the comment still reads broken. maybe the solution is to add `:` in the end